modules/dnn/src/opencl/region.cl

```diff
@@ -84,9 +85,9 @@ __kernel void softmax_activ(const int count,
             output[i] = e;
         }
 
-        int y = index / anchors / cols;
-        int x = index / anchors % cols;
-        int a = index - anchors * (x + y * cols);
+        int y = (index % boxes_per_sample) / anchors / cols;
+        int x = (index % boxes_per_sample) / anchors % cols;
+        int a = (index % boxes_per_sample) - anchors * (x + y * cols);
```

Comment:
It could be replaced to
```cpp
        int y = (index / (anchors * cols)) % rows;
        int x = (index / anchors) % cols;
        int a = index % anchors;
```